parser.go

```diff
@@ -0,0 +1,68 @@
+package telegraf
+
+// Parser is an interface defining functions that a parser plugin must satisfy.
+type Parser interface {
+	// Parse takes a byte buffer separated by newlines
+	// ie, `cpu.usage.idle 90\ncpu.usage.busy 10`
+	// and parses it into telegraf metrics
+	//
+	// Must be thread-safe.
+	Parse(buf []byte) ([]Metric, error)
+
+	// ParseLine takes a single string metric
+	// ie, "cpu.usage.idle 90"
+	// and parses it into a telegraf metric.
+	//
+	// Must be thread-safe.
+	ParseLine(line string) (Metric, error)
+
+	// SetDefaultTags tells the parser to add all of the given tags
+	// to each parsed metric.
+	// NOTE: do _not_ modify the map after you've passed it here!!
+	SetDefaultTags(tags map[string]string)
+}
+
+// HeaderParser is an optional interface for parsers that require
+// parsing of header-information. This is relevant for input plugins
+// using the ParseLine function.
+type HeaderParser interface {
```

Comment:
It looks like the CSV parser is the only parser that parses header information, is there a parser I overlooked that would benefit from this interface as well ?